tests/Http/AcceptHeaderTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Spiral Framework.
+ *
+ * @license   MIT
+ * @author    Pavel Z
+ */
+
+declare(strict_types=1);
+
+namespace Spiral\Http\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Spiral\Http\Header\AcceptHeader;
+use Spiral\Http\Header\AcceptHeaderItem;
+
+class AcceptHeaderTest extends TestCase
+{
+    public function testHeaderItemCompare()
+    {
+        $this->assertEquals(0, AcceptHeaderItem::compare('', ''));
+        $this->assertEquals(0, AcceptHeaderItem::compare('test', 'test2'));
+        $this->assertEquals(1, AcceptHeaderItem::compare('test', 'test2;q=0.7'));
+        $this->assertEquals(-1, AcceptHeaderItem::compare('test; q=0.7', 'test2'));
+
+        $this->assertEquals(0, AcceptHeaderItem::compare('test; q=0.7', 'test2; q=0.7'));
+        $this->assertEquals(0, AcceptHeaderItem::compare('test; q=0.7', 'test2; q=0.7;p'));
+        $this->assertEquals(0, AcceptHeaderItem::compare('test; q=0.7; test=23', 'test2; q=0.7;p=1'));
+        $this->assertEquals(1, AcceptHeaderItem::compare('test; q=0.7; p=2; st=3', 'test2; q=0.7;p=1'));
+        $this->assertEquals(-1, AcceptHeaderItem::compare('test; q=0.7', 'test2; q=0.7;p=1'));
+
+        $this->assertEquals(1, AcceptHeaderItem::compare(
+            AcceptHeaderItem::fromString('test'),
+            AcceptHeaderItem::fromString('*')
+        ));
+    }
+
+    public function testCompareHeaderItemValue()
```

Comment:
I don't think this is a good test case. Testing private methods shows that the abstraction is badly designed, you could add testable items into the accept header and assert the sorted output instead